### PR TITLE
Playwright: The great block inserter refactor showdown.

### PR DESCRIPTION
**Pull request by dpasque**
#### Changes proposed in this Pull Request

Golly gee, that's a lot of files!! 😱 

This PR makes refactor that had a bit of a "thread pulling" effect as we had some assumptions in our block smoke testing that consequently needed updating.

To help simplify review, the real substantial changes that happen here are...
1. We pull the block inserting methods out of `EditorGutenbergComponent` into their own component `EditorSidebarBlockInserterComponent`.
2. We make a new component for inline block inserting, called the `EditorInlineBlockInserterComponent`.
3. We add both of these to the `EditorPage`, and break out the block/pattern adding methods to support both adding from sidebar and inline.
4. We update the `EditorContext` used in block smoke testing flows to no longer be dependent on having the editor Iframe, but instead to have a pointer to the editor `Locator` and an instance of the `EditorPage` itself. This allows us to match the new patterns introduced by the `EditorPage` refactor.

And the subsequent minor changes that follow and add a ton of files...
5. As part of the `EditorContext` change, we move all calls in the block flows to use the `Locator` API.
6. We update other test scripts to use the new block insertion methods.

#### Testing instructions

- [x] Gutenberg desktop passes
- [x] Gutenberg mobile passes
- [x] Pre-release tests pass
- [x] PR desktop passes
- [x] PR mobile passes
- [x] Coblocks desktop passes
- [x] Coblock mobile passes

### CODE REVIEW

**Comment by dpasque** [comment]
This needs a big rebase before review!

File: packages/calypso-e2e/src/lib/pages/editor-page.ts
Comment (dpasque) [inline packages/calypso-e2e/src/lib/pages/editor-page.ts:37]: I don't think we need to export this and apply it to the components. Since TypeScript supports duck typing (of sorts), if the signatures of those component classes changes so that we can't consolidate them here with this interface, we'll get typing errors!

**Comment by dpasque** [comment]
The switch to the locator API is showing a lot of cases where we don't have specificity in our block flows -- I'll revisit each of those and make them all more specific before merge!

File: packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts:85]: I assume the 'button' here refers to the 'add block' button?

File: packages/calypso-e2e/src/lib/blocks/block-flows/payments.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/blocks/block-flows/payments.ts]: Note, `locator.isXYZ` methods are borderline deprecated (if not already). See the comment on the `timeout` parameter [here](https://playwright.dev/docs/api/class-locator#locator-is-visible).

I think we might be better off using a `try/catch` with `waitFor` here, or a `count` if steps are taken to wait for the page  to settle beforehand.

File: packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts]: I can see myself or others being confused by the use of Delegate here, and in `EditorPage`. Perhaps a short comment explaining our intent in either location would be great to have.

**Review by worldomonation — APPROVED** [review]
I have no reasons to withhold approval as I think you've done a great work here. The changes proposed make sense and it's easy to understand why new parts of the code is required.

File: packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts
Comment (dpasque) [reply packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts:85]: Yep! 👍 

Let me know if you think that could use some clarification in the code!

File: packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts
Comment (dpasque) [reply packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts]: Sure thing! I'll put comments in the scripts themselves.

As a quick gut check -- do you think it's too confusing to even take a delegate as a parameter?

I waffled back and forth a lot here... Adding blocks inline is tricky, because there's so many different ways to launch the inserter! And you can't even reliably guarantee there to be only one add block button on the page 😆 .

However, that makes the `addBlockInline` method notably different from `addBlockFromSidebar`, which can safely take care of launching the sidebar. I was trying to find the most obvious way to make that distinction clear, and adding a required delegate to the signature seemed the most intrusive, but also safest, route.

**Comment by fullofcaffeine** [comment]
@dpasque Awesome work! 😍 

>  dependent on having the editor Iframe

This is currently causing failures in the AT builds. Do you mind if I fix the conflicts and merge this?

cc @worldomonation

File: packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts]: I think I was being more confused by the term "Delegate" more than anything. I had a very superficial understanding of what a delegate does in the software context, but have never encountered it or knowingly used it. 

To make sure I understand correctly here: the delegate here refers to any inline block inserter, be it on the main document or within a block. The idea of this is to say that 'whatever happens, something that implements an inline inserter will be passed in here'. Is that more or less correct?

Re: the challenges of the Inline inserter - this is definitely an area that warrants further investigation if we continue developing increasing complex tests.

Comment (dpasque) [reply packages/calypso-e2e/src/lib/blocks/block-flows/layout-grid.ts]: Ahh, after some internet research, I'm now realizing that "delegate" is a term I picked up from my old c# days, and that it's not as universal a term! 

It's even simpler: a "delegate" is just a function passed as a parameter! So, a callback basically (but a callback kind of implies that the function is called at the end).

I'm going to remove that verbiage. Sorry for the confusion!

**Pull request merged**
